.pipelines/templates/template-podman.yml

```diff
@@ -0,0 +1,174 @@
+          UNIQUE_TAG="${IMAGE_TAG}-$(date +%Y%m%d%H%M%S)-$(Build.BuildId)"
+          podman --remote --url=tcp://localhost:8888 images
+
+          echo "Captured UNIQUE_TAG: $UNIQUE_TAG"
+
+          # Retag the image with the ACR repository and unique tag
+          echo "Tagging the image with ACR repository..."
+          podman --remote --url=tcp://localhost:8888 tag localhost/${IMAGE_REPO_NAME}:${IMAGE_TAG} ${ACR_FQDN}/${REPOSITORY}:${UNIQUE_TAG}
+
+          # Push the image to ACR
+          echo "Pushing image to ACR..."
+          podman --remote --url=tcp://localhost:8888 push ${ACR_FQDN}/${REPOSITORY}:${UNIQUE_TAG}
+
+          # Optionally tag and push the image as 'latest'
+          if [ ${{ parameters.pushLatest }} == true ]; then
+            echo "Tagging and pushing latest image..."
+            podman --remote --url=tcp://localhost:8888 tag ${ACR_FQDN}/${REPOSITORY}:${UNIQUE_TAG} ${ACR_FQDN}/${REPOSITORY}:latest
+            podman --remote --url=tcp://localhost:8888 push ${ACR_FQDN}/${REPOSITORY}:latest
+          fi
+
+          # Check for images with label 'stage=rp-build-cache-layer' to extract cover.out
+          echo "Checking for images with label 'stage=rp-build-cache-layer'..."
+          RP_IMAGE_ID=$(podman image ls --filter label=stage=rp-build-cache-layer --noheading --format '{{.Id}}' | tail -n 1)
+
+          if [ -n "$RP_IMAGE_ID" ]; then
+              # If the image exists, extract cover.out
+              echo "Found RP image: $RP_IMAGE_ID"
+              podman create --name extract_cover_out $RP_IMAGE_ID
+              podman cp extract_cover_out:/app/cover.out $(Build.SourcesDirectory)/cover.out
+              podman rm extract_cover_out
+
+              # Ensure cover.out was copied
+              if [ ! -f $(Build.SourcesDirectory)/cover.out ]; then
+                  echo "Error: cover.out file was not extracted successfully."
+                  exit 1
+              else
+                  echo "cover.out file found in $(Build.SourcesDirectory)"
+              fi
+
+              # Validate and display cover.out contents
+              echo "Contents of cover.out:"
+              cat $(Build.SourcesDirectory)/cover.out
+
+              # Check if the file is not empty
+              if [ -s $(Build.SourcesDirectory)/cover.out ]; then
+                echo "cover.out contains data:"
+                head -n 10 $(Build.SourcesDirectory)/cover.out
+              else
+                echo "cover.out is empty or does not exist."
+                exit 1
+              fi
+
+              # Convert cover.out to human-readable format (for Go coverage)
+              echo "Converting cover.out to human-readable format..."
+              go tool cover -func $(Build.SourcesDirectory)/cover.out
+          else
+              # If no image is found, skip the cover.out extraction
+              echo "No RP Image found with label 'stage=rp-build-cache-layer'. Skipping cover.out extraction."
```

Comment:
This looks like a near-dupe of what `make-ci` does. Instead of printing the contents to stdout, let's use the `task: PublishCodeCoverageResults@1` as used in the noncontainerized build process. This integrates our unit test results with ADO directly so we get a UI representation of passes/failures and can detect things like flaky tests over time/etc.